src/components/AvailabilityGrid.tsx

```diff
@@ -0,0 +1,155 @@
+    setGrid(initialGrid())
+  }, [daysOfWeek.length, timeArray.length])
+
+  // Function is called when the mouse is pressed down on a cell
+  const handleMouseDown = (rowIndex: number, colIndex: number) => {
+    if (!isAvailable) return
+    console.log(`Mouse Down: Row ${rowIndex}, Col ${colIndex}`)
+    setIsSelecting(true)
+    toggleCell(rowIndex, colIndex)
+  }
+
+  /* 
+    Function is called when the mouse enters a cell while 
+    the mouse button is pressed 
+  */
+  const handleMouseEnter = (rowIndex: number, colIndex: number) => {
+    if (!isAvailable) return
+    if (isSelecting) {
+      console.log(`Mouse Enter: Row ${rowIndex}, Col ${colIndex}`)
+      toggleCell(rowIndex, colIndex)
+    }
+  }
+
+  // Function is called when the mouse button is released
+  const handleMouseUp = () => {
+    console.log('Mouse Up')
+    setIsSelecting(false)
+  }
+
+  // Function toggles the value of a cell
+  const toggleCell = (rowIndex: number, colIndex: number) => {
+    console.log(`Toggling Cell: Row ${rowIndex}, Col ${colIndex}`)
+    const newGrid = [...grid]
+    newGrid[rowIndex][colIndex] = !newGrid[rowIndex][colIndex]
+    setGrid(newGrid)
+  }
+
+  return (
+    <div onMouseUp={handleMouseUp} onMouseLeave={handleMouseUp}>
+      <div
+        className="grid-container"
+        style={{ display: 'grid', gridTemplateColumns: 'auto 1fr' }}
+      >
+        {/* Time row headers */}
+        <div
+          className="grid-time-headers"
+          style={{
+            display: 'grid',
+            gridTemplateRows: `repeat(${dimensions.height}, 1fr)`,
+            alignItems: 'center',
+          }}
+        >
+          {timeArray.map((time, index) => (
+            <div
+              key={index}
+              className="flex items-center justify-end border-gray-300 pr-2"
+              style={{ height: '64px' }}
+            >
+              {time}
+            </div>
+          ))}
+        </div>
+
+        <div>
+          {/* Grid Column header displaying days of the week */}
+          <div
+            className="grid-header"
+            style={{
+              display: 'grid',
+              gridTemplateColumns: `repeat(${dimensions.width}, 1fr)`,
+            }}
+          >
+            {daysOfWeek.map((day, index) => (
+              <div
+                key={index}
+                className="flex items-center justify-center border-gray-300"
+                style={{ height: '2rem' }}
+              >
+                {day}
+              </div>
+            ))}
+          </div>
+
+          {/* Main Grid body cells for selecting and deselecting */}
+          <div
+            className="grid h-full border border-gray-300"
+            style={{
+              display: 'grid',
+              gridTemplateColumns: `repeat(${dimensions.width}, 1fr)`,
+              gridTemplateRows: `repeat(${dimensions.height}, 1fr)`,
+              width: '100%',
+              height: `${dimensions.height * 64 + 3}px`,
```

Comment:
whenever i see constants I always like to see a comment to why. Why 64? why "+ 3" ?

Reply:
Working on this, comments for this will be added in the next commit